controllers/awsmachinetemplate_controller.go

```diff
@@ -151,16 +152,28 @@ func (r *AWSMachineTemplateReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 				return ctrl.Result{}, err
 			}
 			controllerutil.RemoveFinalizer(awsCluster, key.FinalizerName(iam.ControlPlaneRole))
-			err = r.Update(ctx, awsCluster)
+			patchHelper, err := patch.NewHelper(awsMachineTemplate, r.Client)
```

Comment:
is it possible to create the `patchHelper` just once and re use it instead of creating it twice in line 155 and line 168?